src/fswAlgorithms/orbitControl/lambertSolver/lambertSolver.h

```
@@ -0,0 +1,52 @@
+/*
+ ISC License
+
+ Copyright (c) 2023, Autonomous Vehicle Systems Lab, University of Colorado Boulder
+
+ Permission to use, copy, modify, and/or distribute this software for any
+ purpose with or without fee is hereby granted, provided that the above
+ copyright notice and this permission notice appear in all copies.
+
+ THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+*/
+
+
+#ifndef LAMBERTSOLVER_H
+#define LAMBERTSOLVER_H
+
+#include "architecture/_GeneralModuleFiles/sys_model.h"
+#include "architecture/msgPayloadDefC/LambertProblemMsgPayload.h"
+#include "architecture/msgPayloadDefC/LambertSolutionMsgPayload.h"
+#include "architecture/msgPayloadDefC/LambertPerformanceMsgPayload.h"
+#include "architecture/utilities/bskLogging.h"
+#include "architecture/messaging/messaging.h"
+#include "architecture/utilities/avsEigenSupport.h"
+#include "architecture/utilities/astroConstants.h"
+#include <vector>
+
+/*! @brief This module solves Lambert's problem using either the Gooding or the Izzo algorithm.
+ */
+class LambertSolver: public SysModel {
+public:
+    LambertSolver();
+    ~LambertSolver();
+
+    void Reset(uint64_t CurrentSimNanos);
```

:small_red_triangle_down: Let's mark these as `override`, like `void Reset(uint64_t CurrentSimNanos) override;` (same for Update). It tells the compiler that we are explicitly overriding the abstract  and thee function signature should match exactly. If we don't and get the function signature slightly wrong it will compile and can lead to silent errors.

---

Done.